clash-lib/src/Clash/Backend/VHDL.hs

```diff
@@ -1430,6 +1438,11 @@ patLitCustom _ x y = error $ $(curLoc) ++ unwords
   [ "You can only pass CustomSP / CustomSum and a NumLit to this function,"
   , "not", show x, "and", show y]
 
+renderCompDecls :: [(TextS.Text, [(TextS.Text, PortDirection, HWType)])] -> VHDLM Doc
+renderCompDecls [] = emptyDoc
+renderCompDecls (c:cs) =
+    renderCompDecl c <> line <> line <> renderCompDecls cs
```

Comment:
That way you don't need this special component-specific function, you can just use `insts`